src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -2538,6 +2538,161 @@ class C
             Assert.Equal(expectedText, text);
         }
 
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemove_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"
+class C
+{
+// before
+void M()
+{
+} // after
+}");
+
+            var expectedText = @"
+class C
+{
+
+}";
+
+            var m = cu.DescendantNodes().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal(expectedText, text);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveWithoutEOL_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"class A { } class B { } // test");
+
+            var m = cu.DescendantNodes().OfType<TypeDeclarationSyntax>().LastOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal("class A { } ", text);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveBadDirectiveWithoutEOL_KeepEndOfLine_KeepDirectives()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"class A { } class B { } #endregion");
+
+            var m = cu.DescendantNodes().OfType<TypeDeclarationSyntax>().LastOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine | SyntaxRemoveOptions.KeepDirectives);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal("class A { } #endregion", text);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveDocument_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"
+#region A
+class A 
+{ } 
+#endregion");
+
+            var cu2 = cu.RemoveNode(cu, SyntaxRemoveOptions.KeepEndOfLine);
+
+            Assert.Null(cu2);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveFirstParameter_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"
+class C
+{
+void M(
+// before a
+int a,
```

Comment:
Could you add some trailing trivia for the parameter in this test?

Reply:
Added in `TestRemoveFirstParameterEOLCommaTokenLeadingTrivia_KeepEndOfLine` and `TestRemoveLastParameterEOLCommaTokenLeadingTrivia_KeepEndOfLine` tests.